### ISSUE TITLE
Error handling for invalid local resource configuration

### ISSUE THREAD

**Issue by timmattison** [awslabs/aws-greengrass-provisioner#160]
There are a few ways that local resources can cause the provisioner to fail:

- 1 - A local resource is specified in the configuration that doesn't exist
- 2 - A local resource is specified in the configuration that can not be accessed by the user running Greengrass

An example of the error seen in case 2:

```
Greengrass service reported an error [Deployment 20152727-d68e-4384-a8f2-e704712f9b23 of type NewDeployment for group a7d23862-5e86-43c3-9347-cc11fe77edc1 failed error: Greengrass deployment error: unable to execute download step in deployment. error while processing: unable to load the group file downloaded: resource access policy is invalid, functionArn: arn:aws:lambda:us-east-1:zzz:function:fff-bbb:2, resource: /aaa/ddd/eee: user or group doesn't have permission on the file, userId: 109, userGroupId: [995 0], access: rw, filePath: /aaa/ddd/eee]
```

When case 2 occurs the provisioner goes into a loop disassociating and reassociating the role with the group because it is generically assumed to be an error related to the role not being available yet.

**Comment by OscarVanL** [comment]
I'm having this error, and it's caused by case 2. How exactly do I give the user running Greengrass permission to access this resource?
I'm adding a Local volume resource, /home/pi/input, I've checked the "Automatically add OS group permissions of the Linux group that owns the resource." box and I've set read and write permissions, but when I press Deploy I get error:

Deployment ----------------- of type NewDeployment for group ------------- failed error: Greengrass deployment error: unable to execute download step in deployment. error while processing: unable to load the group file downloaded: resource access policy is invalid, functionArn: ----------, resource: /home/pi/input: user or group doesn't have permission on the file, userId: 111, userGroupId: [116 1000], access: rw, filePath: /home/pi/input
